csharp/ql/lib/semmle/code/csharp/frameworks/Razor.qll

```diff
@@ -215,3 +215,61 @@ private class RelativeViewCallFilepath extends NormalizableFilepath {
   /** Holds if this string is the `idx`th path that will be searched for the `vc` call. */
   predicate hasViewCallWithIndex(ViewCall vc, int idx) { vc = vc_ and idx = idx_ }
 }
+
+/** A subclass of `Microsoft.AspNetCore.Mvc.RazorPages.PageModel` */
+class PageModelClass extends Class {
+  PageModelClass() {
+    this.getABaseType+().hasFullyQualifiedName("Microsoft.AspNetCore.Mvc.RazorPages", "PageModel")
+  }
+
+  /** Gets a handler method such as `OnGetAsync` */
+  Method getAHandlerMethod() {
+    result = this.getAMethod() and
+    result.getName().matches("On%") and
```

Comment:
Should `OnPageHandler%` methods be filtered out?

Reply:
Hmm, I guess those methods are not technically handler methods; however they are called during the page handler process, so it makes sense to consider tainted writes to the model in those methods to still propagate to the page.